From c2bd4817d440b422c5fa2fc2c375e10a736e02f2 Mon Sep 17 00:00:00 2001
From: Zhouwang Huang <honjow311@gmail.com>
Date: Wed, 8 Apr 2026 16:56:27 +0800
Subject: [PATCH] HID: hid-oxp: initialize button map defaults before delayed
 init work

The bmap pages are allocated with devm_kzalloc (zero-initialized) but
oxp_set_defaults_bmap_1/2 is never called before the delayed init work
fires. This causes oxp_set_buttons() to send all-zero mapping indexes,
mapping every button to oxp_button_table[0] (BTN_A).

Call oxp_set_defaults_bmap_1/2 right after allocation so the init work
has correct default values.
---
 drivers/hid/hid-oxp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/hid/hid-oxp.c b/drivers/hid/hid-oxp.c
index cad6973089a0..a40ed418b9eb 100644
--- a/drivers/hid/hid-oxp.c
+++ b/drivers/hid/hid-oxp.c
@@ -1497,6 +1497,8 @@ static int oxp_cfg_probe(struct hid_device *hdev, u16 up)
 
 	drvdata.bmap_1 = bmap_1;
 	drvdata.bmap_2 = bmap_2;
+	oxp_set_defaults_bmap_1(bmap_1);
+	oxp_set_defaults_bmap_2(bmap_2);
 	drvdata.rumble_intensity = 5;
 	mod_delayed_work(system_wq, &oxp_mcu_init, msecs_to_jiffies(50));
 
-- 
2.51.2

